### DISCUSSION TITLE
Creating gifs from photos in sub-folders

### DISCUSSION THREAD

**Discussion by BlackLoneWolf** [ImageMagick/ImageMagick#4018] [Help] [answered]
Hi, I want to create gifs from jpg photos. I have a main folder in which I have several hundred smaller folders with several pictures. For each folder I need one gif with the pictures inside.
I have tried firing up imagemagick in the main folder and using the code

for /r "G:\Nowy folder (2)" %a in (*.*) do convert -delay 20 -loop 0 *.jpg myimage.gif

But only error pops up. 
"invalid parameter -20"

OS - Windows 10
Thank you in advance for your help.

**Comment by snibgo** [comment]
What version of IM? If v7, I suggest you use "magick" instead of "convert".

> "invalid parameter -20"

That means you are not running the ImageMagick "convert" program. Instead, you are running the Microsoft Windows "convert" program.

Your "for" command contains another error: you loop through directories, assigning the directory name to %a, but you never use %a in your "convert" command.

**Reply by BlackLoneWolf** [reply]
First, I'm a total newbie to ImageMagick and I tried to do this by gluing together two different codes from github
I use version v.7.1
Now I write: 
"for /r "G:\Nowy folder (2)" %a in (.) do magick -delay 20 -loop 0 *.jpg myimage.gif" or
"for /r /d %a in (*) do magick -delay 20 -loop 0 *.jpg myimage.gif"
and i have another problem
"magick: unable to open image '*.jpg': Invalid argument @ error/blob.c/OpenBlob/3537."

**Comment by snibgo** [comment, marked as answer]
When pasting code into the forum, please use the markup tools provided. Or use a line of three backticks at the start and at the end. Otherwise, characters like \* get stripped, and posts are difficult to understand.

I suggest:
```
for /r "G:\Nowy folder (2)" %a in (.) do magick -delay 20 -loop 0 %a\*.jpg %a\myimage.gif
```

But I suggest that you work out the "magick" command first. When it does what you want, put it in a "for" loop.

**Reply by BlackLoneWolf** [reply]
Thanks, now it's working.
One additional comment.
If it doesn't work for anyone, please check if your folders are named with one word names. If not, your gifs will not be created.

**Reply by snibgo** [reply]
If you have spaces in directories or filenames, you need to quote them:
```
for /r "G:\Nowy folder (2)" %a in (.) do magick -delay 20 -loop 0 "%a\*.jpg" "%a\myimage.gif"
```
... but I think spaces are a really bad idea.
